src/golang/lib/lib_utils/utils.go

```diff
@@ -84,21 +84,33 @@ func RunCmd(command string, args []string, dir string, stream bool) (string, str
 		stdoutScanner := bufio.NewScanner(stdout)
 		stderrScanner := bufio.NewScanner(stderr)
 
+		var stderrMsg string
+		// Create a channel to communicate between the main process and the goroutine to exchange
+		// the stderr message.
+		ch := make(chan string)
+
 		// start separate goroutines to stream the output from each scanner
 		go func() {
 			for stdoutScanner.Scan() {
 				log.Infof("stdout: %s", stdoutScanner.Text())
 			}
 		}()
 		go func() {
+			var sb strings.Builder
 			for stderrScanner.Scan() {
```

Comment:
I don't know too much about the bufio.Scanner class, but this is guaranteed to return even if nothing is ever logged to stderr? How does it know that...

Reply:
Yes, when the program exits, the scanner will reach the end and break out of the for loop.